### DISCUSSION TITLE
Crontab и Действия по расписанию

### DISCUSSION THREAD

**Discussion by UPRLS9000** [totumonline/totum-ru-issues-and-discussions#592] [Как это сделать?] [answered]
Здравствуйте.
Подскажите, пожалуйста, что указать в задании для Crontab, чтобы работали Кроны/Действия по расписанию. Видел где то в документации или в видео, что в задании нужно указать определенный файл, но сейчас не могу найти.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Суть вопроса непонятна. Если все установлено скриптом установки то в crontabe все прописано. Далее задания по расписанию управляются в таблице crons.

В документации: https://ru.docs.totum.online/crons#crons

**Comment by UPRLS9000** [comment]
Totum сейчас стоит на обычном хостинге, перенесен с другого сервера, кроны не настроены.
Подскажите, пожалуйста, что сюда нужно указать:
![image](https://github.com/user-attachments/assets/d698e3f4-2147-434e-b017-a04ad37c0128)

**Comment by ttm-alexeypolunin** [comment]
Стандартные кроны такие


```
* * * * * cd $HOME/totum-mit/ && bin/totum schemas-crons
*/10 * * * * cd $HOME/totum-mit/ && bin/totum clean-tmp-dir
*/10 * * * * cd $HOME/totum-mit/ && bin/totum clean-schemas-tmp-tables
```

как у вас установлено неизвестно

**Comment by UPRLS9000** [comment]
Запуская  по ssh вручную:
`/usr/bin/php8.2 bin/totum schemas-crons`

Получаю в ответ:
```
Command "schemas-crons" is not defined.

  Did you mean one of these?
      schema-cron
      schema-crons
```
schema-crons - никакой реакции

schema-cron с указанием id крона из таблицы кронов работает и действия выполняются, в чем может быть проблема? Неверно указываю команду? Подскажите, что я делаю не так? 

totum, V 4.11.57.0

**Reply by ttm-alexeypolunin** [reply]
это старый формат установки, когда на сервере только одна схема

используйте этот формат кронов

schemas -> schema

```
* * * * * cd $HOME/totum-mit/ && bin/totum schema-crons
*/10 * * * * cd $HOME/totum-mit/ && bin/totum clean-tmp-dir
*/10 * * * * cd $HOME/totum-mit/ && bin/totum clean-schema-tmp-tables
```

**Comment by UPRLS9000** [comment]
Так и не удалось заставить работать:
`* * * * * cd $HOME/totum-mit/ && bin/totum schema-crons`

Просто ничего не происходит, похоже я жестко туплю)

Но всё равно спасибо вам за помощь.

**Reply by ttm-alexeypolunin** [reply]
Тк у вас установлено не нашим установщиком, то мы не знаем в какой папке на сервере расположена система. Адаптируйте пути под ваш случай.

**Reply by UPRLS9000** [reply]
я адаптировал.

Срабатывает, если я запускаю конкретный крон по id:
```
bin/totum schema-cron 1
bin/totum clean-tmp-dir
bin/totum clean-schema-tmp-tables
```

Но если пытаюсь выполнить все сразу, как вы написали выше, то ничего не происходит:
```
bin/totum schema-crons
bin/totum clean-tmp-dir
bin/totum clean-schema-tmp-tables
```

![image](https://github.com/user-attachments/assets/5232c91e-db3b-42b9-b8ce-bee18f24ba90)

**Reply by ttm-alexeypolunin** [reply]
Наша рекомендация — реинсталировать на vps скриптом установки.
